Assets/MixedRealityToolkit/_Core/Devices/UnityInput/UnityTouchDeviceManager.cs

```diff
@@ -0,0 +1,305 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Devices;
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Physics;
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Devices;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Physics;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.TeleportSystem;
+using Microsoft.MixedReality.Toolkit.Core.Managers;
+using Microsoft.MixedReality.Toolkit.Core.Utilities;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Devices.UnityInput
+{
+    /// <summary>
+    /// Manages Touch devices using unity input system.
+    /// </summary>
+    public class UnityTouchDeviceManager : BaseDeviceManager
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="priority"></param>
+        public UnityTouchDeviceManager(string name, uint priority) : base(name, priority) { }
+
+        /// <summary>
+        /// Internal Touch Pointer Implementation.
+        /// </summary>
+        private class TouchPointer : IMixedRealityPointer
+        {
+            /// <summary>
+            /// Constructor.
+            /// </summary>
+            /// <param name="pointerName"></param>
+            public TouchPointer(string pointerName)
+            {
+                InputSystem = MixedRealityManager.Instance.GetManager<IMixedRealityInputSystem>();
+                PointerId = InputSystem.FocusProvider.GenerateNewPointerId();
+                PointerName = pointerName;
+            }
+
+            /// <inheritdoc />
+            public IMixedRealityInputSystem InputSystem { get; }
```

Comment:
The BaseDeviceManager already provides this, so it's a little redundant here

Reply:
It's part of the pointer interface, so can't really change that.